pkgs/applications/editors/emacs/build-support/melpa.nix

```diff
@@ -98,9 +97,21 @@ libBuildHelper.extendMkDerivation' genericBuild (finalAttrs:
 
   preUnpack = ''
     mkdir -p "$NIX_BUILD_TOP/recipes"
-    if [ -n "$recipe" ]; then
-      cp "$recipe" "$NIX_BUILD_TOP/recipes/$ename"
+    recipeFile="$NIX_BUILD_TOP/recipes/$ename"
```

Comment:
I usually treat Bash as a lower level language.
I avoid it when the same thing can be done with Nix/Nixpkgs machinery.

If I understood this correctly, this code tries to detect  if the recipe is a file or a string.
If file, pick it as is.
If string, put the string contents into a file.

This is similar to postPatch hook from `dwl`:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/dw/dwl/package.nix#L78-L86

Reply:
Hmm, the point is to support not using a derivation for recipe, as @adisbladis said in https://github.com/NixOS/nixpkgs/pull/330589#discussion_r1717900614.

Reply:
Hum, I do not know if this is the idea from adisbladis in that comment.

To me it looked like "the user (whether human or machine) should have the option to pass a file or a string" (and, of course, files can be generated by derivations too).

In this sense the example I provided looks cleaner and reduces the Bash code to a couple lines.
The Nix code receives the recipe in whatever format, and returns a file containing the recipe, so that the Bash code does nothing besides linking that file.

The only advantage of using the Bash code I can think is maybe some micro optimization.

Reply:
I think the goal to avoid using a derivation for recipe.  So in your example, [this][1] should be avoided.

[1]: https://github.com/NixOS/nixpkgs/blob/6f8ad7e4d0bcb4e93aced92ebe7a9b45a8150a26/pkgs/by-name/dw/dwl/package.nix#L84

Reply:
It does make any difference? In the end we have a string converted to a file, whether if it is made by Bash or Nix+Nixpkgs.

I think the Nix code is slightly more readable.
On the other hand, if the Bash code is more efficient (maybe because Nix is slower, or some context in the background has a hidden penalty), then it would be something positive.

Reply:
I guess not using a derivation increases performances a bit.  To benchmark that, someone needs to make generated melpa packages use recipe content string first.